onnx/checker.py

```diff
@@ -38,6 +39,35 @@ def check_node(node):
             'NodeProto of type {} did not pass defs schema check.'.format(str(node.op_type)))
 
 
+def check_tensor_value_info(value_info,
+                            type_required=True,
+                            shape_required=True):
+    if not isinstance(value_info, ValueInfoProto):
+        raise RuntimeError('You cannot pass an object that is not ValueInfoProto.')
+    if not value_info.name:
+        raise NameError('ValueInfoProto must have its name set.')
+
+    if not type_required and not shape_required:
```

Comment:
you need to gate only the type part with `type_required`, similarly for `shape`

Reply:
@dzhulgakov Both elem_type and shape are stored together in the `type` field.